### PR TITLE
Have Hibernate Reactive depend on quarkus-mutiny correctly

### LINKED ISSUE #15111: Hibernate Reactive component is missing the dependency to Quarkus-Mutiny
**Describe the bug**
Hibernate Reactive depends on Mutiny, but isn't using the `quarkus-mutiny` module; this implies it's missing the deployment module for Mutiny as well.

### PR DESCRIPTION
**Pull request by Sanne**
Fixes #15111

### CODE REVIEW

File: extensions/hibernate-reactive/runtime/pom.xml [hunk 1]
```diff
@@ -34,6 +34,17 @@
         <dependency>
             <groupId>org.hibernate.reactive</groupId>
             <artifactId>hibernate-reactive-core</artifactId>
+            <exclusions>
+                <!-- Make sure Quarkus can manage Mutiny via quarkus-mutiny -->
+                <exclusion>
+                    <groupId>io.smallrye.reactive</groupId>
+                    <artifactId>mutiny</artifactId>
```
Comment (yrodiere) [inline extensions/hibernate-reactive/runtime/pom.xml:41]: Is this really necessary? I'd expect the dependency management in the BOM to be enough.

**Review by yrodiere — APPROVED** [review]

File: extensions/hibernate-reactive/runtime/pom.xml (hunk 1 above)
Comment (gsmet) [reply extensions/hibernate-reactive/runtime/pom.xml:41]: Yeah, the exclusion is not needed but doesn't hurt either.

**Comment by Sanne** [comment]
Yes, the BOM should be enough, but users don't always use our BOM.

Shouldn't matter also as we're hoping to keep the versions aligned but this
should be safer.

**Comment by Sanne** [comment]
@gsmet I can't merge it because Yoann's review isn't authoritative enough.. let's fix that at the next round?

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
I will do 2 more rounds of backports so it will be included.

**Pull request merged**